src/App.tsx

```diff
@@ -21,55 +21,120 @@ enum SortType {
   LENGTH,
 }
 
-// Use this function in the render method
 function getReorderedGoods(
   goods: string[],
   sortType: SortType,
   isReversed: boolean,
 ) {
-  // Not to mutate the original array
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // ...
+  visibleGoods.sort((f1, f2) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return f1.localeCompare(f2);
+
+      case SortType.LENGTH:
+        return f1.length - f2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
 type State = {
   isStarted: boolean,
   isReversed: boolean,
   sortType: SortType,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends React.Component<{}, State> {
+  state = {
+    isStarted: false,
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const visibleGoods = getReorderedGoods(
+      goodsFromServer,
+      this.state.sortType,
+      this.state.isReversed,
+    );
+
+    return (
+      <div className="App">
+        {!this.state.isStarted
+        && (
+          <button
+            type="button"
+            onClick={() => {
+              this.setState(state => ({ isStarted: !state.isStarted }));
+            }}
+          >
+            Start
+          </button>
+        )}
+        {this.state.isStarted
+        && (
+          <div>
+            <button
+              type="button"
+              onClick={() => {
+                this.setState({ sortType: SortType.ALPABET });
+              }}
```

Comment:
create a handler for this, smth like a `sortBy` function which will take a `sortType` param with `SortType` type